app/main.py

```diff
@@ -2,24 +2,76 @@
 
 
 class IntegerRange:
-    pass
+    def __init__(self, min_amount: int, max_amount: int) -> None:
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __set_name__(self, owner: str, name: str) -> None:
+        self.protected_name = "_" + name
+
+    def __get__(self, instance: str, owner: str) -> int:
```

Comment:
The `__get__` method has incorrect type annotations. The `instance` parameter should be of type `object`, not `str`, and `owner` should be of type `type`. Update the type annotations accordingly.